### PR TITLE
CNV-45584: Update Edit CPU | Memory modal UI

### PR DESCRIPTION
**Pull request by pcbailey**
## 📝 Description

This PR updates the design of the CPU/memory modal to give the user two methods for configuring the CPU: enter the CPU topology manually and entering the number of desired vCPUs.

Jira: https://issues.redhat.com/browse/CNV-45584

### Before

cpu-mem-redesign--BEFORE--2024-08-16_22-24

### After

cpu-mem-redesign--AFTER--2024-08-16_21-06

### CODE REVIEW

File: src/utils/components/CPUMemoryModal/components/CPUInput/components/vCPUInput/components/CPUHelperText/CPUHelperText.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+div#cpu-helper-text {
```
Comment (avivtur) [inline src/utils/components/CPUMemoryModal/components/CPUInput/components/vCPUInput/components/CPUHelperText/CPUHelperText.scss:1]: is there a reason why the div is needed? id should be unique

**Review by avivtur — COMMENTED** [review]
Nice work :) 
Left a few comments

File: src/utils/components/CPUMemoryModal/CPUMemoryModal.tsx
[the remark this answers is not in the document]
Comment (pcbailey) [reply src/utils/components/CPUMemoryModal/CPUMemoryModal.tsx:1]: I think I got everything, but please check behind me.

File: src/utils/components/CPUMemoryModal/components/CPUInput/CPUInput.tsx [hunk 2]
```diff
@@ -1,39 +1,103 @@
...
+        {t('CPU')}
+        <Popover
+          bodyContent={t(
+            'As a default, the VirtualMachine CPU uses sockets to enable hotplug. You can also define the topology manually',
+          )}
+        >
+          <Button
+            aria-label="Action"
+            className="cpu-input__title--help-text-button"
+            variant="plain"
```
[the remark this answers is not in the document]
Comment (pcbailey) [reply src/utils/components/CPUMemoryModal/components/CPUInput/CPUInput.tsx]: Done!

File: src/utils/components/CPUMemoryModal/components/CPUInput/CPUInput.tsx [hunk 3]
```diff
@@ -1,39 +1,103 @@
...
+            setSelectedRadioOption={setSelectedRadioOption}
+          />
+        }
+        onClick={() => {
+          setSelectedRadioOption(CPUInputType.editVCPU);
+        }}
+        id={CPUInputType.editVCPU}
+        isChecked={selectedRadioOption === CPUInputType.editVCPU}
+        isLabelWrapped
+        name="cpu-input-type"
```
[the remark this answers is not in the document]
Comment (pcbailey) [reply src/utils/components/CPUMemoryModal/components/CPUInput/CPUInput.tsx]: Done!

File: src/utils/components/CPUMemoryModal/components/CPUInput/components/vCPUInput/components/CPUHelperText/CPUHelperText.scss (hunk 1 above)
Comment (pcbailey) [reply src/utils/components/CPUMemoryModal/components/CPUInput/components/vCPUInput/components/CPUHelperText/CPUHelperText.scss]: At one point in testing it wasn't working without it, but I checked again and it worked fine, so I removed it. =)

File: src/utils/components/CPUMemoryModal/CPUMemoryModal.tsx
[the remark this answers is not in the document]
Comment (avivtur) [reply src/utils/components/CPUMemoryModal/CPUMemoryModal.tsx:1]: Look like you got everything, looks good :+1:

**Pull request merged**